cognite/seismic/protos/v1/seismic_service_messages.proto

```diff
@@ -72,6 +72,47 @@ message DeleteSurveyResponse {
     bool success = 1;
 }
 
+/**
+    Source Files
+**/
+
+message RegisterSourceSegyFileRequest {
+    Identifier survey = 1;                              // [required] The survey this source file belongs to
+    string path = 2;                                    // [required] Path including protocol, bucket and directory structure. Example: "gs://cognite-seismic-eu/samples/"
+    string name = 3;                                    // [required] Unique filename including extension. Example: "DN1302M03R16_MERGED_KPSDM_00-32_DEG_T.sgy". The name must be unique across buckets and can be used to identify this file in query requests
+    ExternalId external_id = 4;                         // [optional] An external identifier - matches service contract field
+    map<string, string> metadata = 5;                   // [optional] Any custom-defined metadata
+    CRS crs = 6;                                        // [required] Official name of the CRS used. Example: "EPSG:23031"
+    SegyOverrides overrides = 7;                        // [optional] Overrides for the source file
+    repeated TraceHeaderField trace_header_fields = 8;  // [optional] The trace header fields that will be used as keys for indexing
+    Dimensions dimensions = 9;                          // [required] File data dimensionality, either 2D or 3D
+}
+
+message RegisterSourceSegyFileResponse {
+    SourceSegyFile file = 1;
+}
+
+message EditSourceSegyFileRequest {
+    Identifier file = 1;                                // [required] The registered source file to edit
+    string path = 2;                                    // [required] Path including protocol, bucket and directory structure. Example: "gs://cognite-seismic-eu/samples/"
+    string name = 3;                                    // [required] Unique filename including extension. Example: "DN1302M03R16_MERGED_KPSDM_00-32_DEG_T.sgy". The name must be unique across buckets and can be used to identify this file in query requests
+    ExternalId external_id = 4;                         // [optional] An external identifier - matches service contract field
+    map<string, string> metadata = 5;                   // [optional] Any custom-defined metadata
+    CRS crs = 6;                                        // [required] Official name of the CRS used. Example: "EPSG:23031"
+    SegyOverrides overrides = 7;                        // [optional] Overrides for the source file
+    repeated TraceHeaderField trace_header_fields = 8;  // [optional] The trace header fields that will be used as keys for indexing
+}
```

Comment:
Should we be able to edit the dimensionality of a file once it's been registered?

Reply:
I was debating on this one.. Both the dimensionality and the trace header fields.
I didn't want to commit to allowing to change the dimensionality at the moment in case it made no sense or allowed the user to put the file in an invalid state. Counting on the fact that it is easier to add a field than to remove one.
If you feel like it should be there, I can put it in.

Reply:
I guess if the user registers the file with the wrong dimensionality, they'd have to go through these steps currently:
* Register file as 3d (should be 2d)
* Ingest file as 3d (either fails or they realize when it completes)
* Delete file, keeping a copy of metadata, crs, etc that's been added
* Re-register the file with the correct dimensionality
* Ingest as 2d

I think it's similar to when they ingest a file with the wrong offsets, which seems to happen fairly often, so it might be good to have it be easier to rectify the mistakes?

Reply:
The way we see the file objects working is that one can edit all the fields without them being reflected in the corresponding seismicstore. So it is perfectly fine to edit the dimensionality, but that will not change the corresponding seismicstore until the file is reingested. There's no invalid state.

So I think we should include editing of the dimensionality and trace header fields.

Reply:
Ok. Done. Also fixed the documentation on what's required/optional